### PR TITLE
[WIP] Top-level API modularization

### PR DESCRIPTION
**Pull request by ShishKabab**
As discussed a few times, this split up the `Ocean` class into `ocean.accounts`, `ocean.assets` and `ocean.sea`. Also, it does retain the top-level methods as aliases to ensure compatibility for existing code. The split is now quick-and-dirty, with the different submodules still using internal stuff of the `Ocean` class. Next step would be moving this stuff to the subcomponents, so the `Ocean` class only has the responsibility to delegate without holding any data.

Also, I've prefixed non-public stuff in the `Ocean` class with `_`, so it's clear what consumers should rely on and what not.

*Still to do*: Agree on the method names in the split out modules! Please take the time to do this before merging, since this affects the long-term cleanliness of the API.

I haven't been able to run the e2e tests because of trouble with Barge. If somebody could run them and everybody agrees about the changes, feel free to merge this without consulting me.

### CODE REVIEW

**Comment by ssallam** [comment]
Instead of passing `ocean` to the new classes, can we just pass a `keeper` instance?

File: squid_py/ocean/ocean_sea.py [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+
+        if not consumer_account.unlock():
+            logger.warning(f'Unlock of consumer account failed {consumer_account.address}')
+
+        agreement_hash = service_agreement.get_service_agreement_hash(agreement_id)
+        signature = consumer_account.sign_hash(agreement_hash)
+
+        service_def = ddo.find_service_by_id(service_definition_id).as_dictionary()
+        # Must approve token transfer for this purchase
+        self._approve_token_transfer(service_agreement.get_price(), consumer_account)
```
Comment (ssallam) [inline squid_py/ocean/ocean_sea.py]: This is invalid now, needs to go through `self._ocean`.

**Comment by josepablofm78** [comment]
Is verify_service_agreement exposed as a public method?

**Comment by ShishKabab** [comment]
> Instead of passing `ocean` to the new classes, can we just pass the instances they actually need?

Yeah, that was the mid-term idea. Because there was some pressure to get this through however, I just did it in a way that I saw with the least risk of bugs. (And still missed some things because of not getting Barge to work.)

> Is verify_service_agreement exposed as a public method?

Doesn't appear so yet. Have to check what came out of the meeting you had internally, to see what needs to be in the public API and where.

File: squid_py/accounts/account.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+"""Ocean module."""
+import logging
+from collections import namedtuple
+
+Balance = namedtuple('Balance', ('eth', 'ocn'))
```
Comment (eruizgar91) [inline squid_py/accounts/account.py]: Maybe this is not needed in this class anymore

Comment (ssallam) [reply squid_py/accounts/account.py]: Removed.

**Pull request merged**